msyncd/synchronizer.cpp

```diff
@@ -1903,10 +1900,22 @@ QStringList Synchronizer::syncProfilesByType(const QString &aType)
     return iProfileManager.profileNames(aType);
 }
 
-void Synchronizer::onNetworkStateChanged(bool aState)
+void Synchronizer::onNetworkStateChanged(bool aState, Sync::InternetConnectionType type)
 {
     FUNCTION_CALL_TRACE;
-    if(!aState) {
+    LOG_DEBUG("Network state changed: OnLine:" << aState << " connection type:" <<  type);
+    if (acceptScheduledSync(aState, type))
+    {
+        LOG_DEBUG("Restart sync for profiles that need network");
+        QStringList profiles(iWaitingOnlineSyncs);
+        iWaitingOnlineSyncs.clear();
+        foreach(QString profileName, profiles)
+        {
+            startSync(profileName, true);
```

Comment:
Are all waiting online syncs also scheduled syncs?  If so, should call startScheduledSync() here, I guess?